src/twisted/mail/test/test_mailmail.py

```diff
@@ -73,3 +109,271 @@ def test_copyAliasedSender(self):
         """
         exc = self.assertRaises(SystemExit, parseOptions, ['-om'])
         self.assertEqual(exc.args, ("Unsupported option.",))
+
+
+    def test_version(self):
+        """
+        The I{--version} option displays the version and raises
+        L{SystemExit}.
+        """
+        out = NativeStringIO()
+        self.patch(sys, 'stdout', out)
+        systemExitCode = self.assertRaises(SystemExit, parseOptions,
+                                           '--version')
+        # SystemExit.code is None on success
+        self.assertEqual(systemExitCode.code, None)
+        data = out.getvalue()
+        self.assertEqual(data, "mailmail version: {}\n".format(version))
+
+
+    def test_backGroundDelivery(self):
+        """
+        The I{-odb} flag specifies background delivery.
+        """
+        stdin = NativeStringIO('\n')
+        self.patch(sys, 'stdin', stdin)
+        o = parseOptions("-odb")
+        self.assertTrue(o.background)
+
+
+    def test_foreGroundDelivery(self):
+        """
+        The I{-odf} flags specifies foreground delivery.
+        """
+        stdin = NativeStringIO('\n')
+        self.patch(sys, 'stdin', stdin)
+        o = parseOptions("-odf")
+        self.assertFalse(o.background)
+
+
+    def test_recipientsFromHeaders(self):
+        """
+        The I{-t} flags specifies that recipients should be obtained
+        from headers.
+        """
+        stdin = NativeStringIO(
+            'To: Curly <invaliduser2@example.com>\n'
+            'Cc: Larry <invaliduser1@example.com>\n'
+            'Bcc: Moe <invaliduser3@example.com>\n'
+            '\n'
+            'Oh, a wise guy?\n')
+        self.patch(sys, 'stdin', stdin)
+        o = parseOptions("-t")
+        self.assertEqual(len(o.to), 3)
```

Comment:
why check only the length and not the actual value?

maybe, due to a bug BCC is parsed as an empty string... and the lenght would still be 3